mlrun/utils/helpers.py

```diff
@@ -474,6 +474,71 @@ def normalize_name(name: str):
     return name.lower()
 
 
+def ensure_batch_job_suffix(
+    function_name: typing.Optional[str],
+) -> tuple[str, bool, str]:
+    """
+    Ensure that a function name has the batch job suffix appended to prevent database collision.
+
+    This helper is used by to_job() methods in runtimes that convert online functions (serving, local)
+    to batch processing jobs. The suffix prevents the job from overwriting the original function in
+    the database when both are stored with the same (project, name) key.
+
+    :param function_name: The original function name (can be None or empty string)
+
+    :return: A tuple of (modified_name, was_renamed, suffix) where:
+        - modified_name: The function name with the batch suffix (if not already present),
+          or empty string if input was empty
+        - was_renamed: True if the suffix was added, False if it was already present or if name was empty
+        - suffix: The suffix value that was used (or would have been used)
+
+    """
+    suffix = mlrun_constants.RESERVED_BATCH_JOB_SUFFIX
+
+    # Handle None or empty string
+    if not function_name:
+        return function_name, False, suffix
+
+    if not function_name.endswith(suffix):
+        return (
+            f"{function_name}{suffix}",
+            True,
+            suffix,
+        )
+    return function_name, False, suffix
+
+
+def strip_batch_job_suffix(
+    function_name: typing.Optional[str],
+) -> tuple[str, bool, str]:
+    """
+    Remove the batch job suffix if present.
+
+    This helper is used when converting a batch job back to a local/serving runtime for execution.
+    The launcher uses this to strip the `-batch` suffix when creating temporary local functions
+    from batch jobs for local execution.
+
+    :param function_name: The function name (may have -batch suffix, can be None or empty string)
+
+    :return: A tuple of (stripped_name, was_stripped, suffix) where:
+        - stripped_name: The function name without the batch suffix (if it was present),
+          or the original name if suffix wasn't present, or empty string/None if input was empty
+        - was_stripped: True if the suffix was removed, False if it wasn't present or if name was empty
+        - suffix: The suffix value that was removed (or would have been removed)
+
+    """
+    suffix = mlrun_constants.RESERVED_BATCH_JOB_SUFFIX
+
+    # Handle None or empty string
+    if not function_name:
+        return function_name, False, suffix
+
+    if function_name.endswith(suffix):
+        return function_name[: -len(suffix)], True, suffix
```

Comment:
I suggest using the more direct [`str.removesuffix`](https://docs.python.org/3/library/stdtypes.html#str.removesuffix) method:
```suggestion
        return function_name.removesuffix(suffix), True, suffix
```

Reply:
will fix